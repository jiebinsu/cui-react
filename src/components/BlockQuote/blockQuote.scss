@import "@coopdigital/foundations-typography/dist/typography.css";
