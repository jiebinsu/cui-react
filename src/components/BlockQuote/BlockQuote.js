import React from "react";
import PropTypes from "prop-types";
import "./blockQuote.scss";

const BlockQuote = ({ className, citation, children }) => {
  return (
    <blockquote>
      <p className={className}>{children}</p>
      <cite>{citation}</cite>
    </blockquote>
  );
};

BlockQuote.defaultProps = {
  className: null,
  citation: null,
  children: null,
};

BlockQuote.propTypes = {
  className: PropTypes.string,
  citation: PropTypes.string,
  children: PropTypes.node,
};

export default BlockQuote;
