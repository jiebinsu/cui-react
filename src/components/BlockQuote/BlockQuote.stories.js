import React from "react";
import BlockQuote from "./BlockQuote";

export default {
  title: "Base/BlockQuote",
  component: BlockQuote,
};

export const Basic = (args) => (
  <BlockQuote {...args}>This is a quote</BlockQuote>
);
Basic.args = {
  citation: "citation",
};
