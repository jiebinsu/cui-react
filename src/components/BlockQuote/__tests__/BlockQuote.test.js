import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import BlockQuote from "../index";

describe("BlockQuote", () => {
  it("should render default BlockQuote", () => {
    const wrapper = mount(<BlockQuote />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<BlockQuote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should work with all variants", () => {
    const Variants = () => (
      <div>
        <BlockQuote className="newClass" citation="cite">
          Quote
        </BlockQuote>
      </div>
    );

    const wrapper = mount(<Variants />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Variants />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
