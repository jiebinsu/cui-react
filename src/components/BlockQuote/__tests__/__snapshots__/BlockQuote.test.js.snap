// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockQuote should render default BlockQuote 1`] = `
<blockquote>
  <p
    className={null}
  />
  <cite />
</blockquote>
`;

exports[`BlockQuote should work with all variants 1`] = `
<div>
  <blockquote>
    <p
      className="newClass"
    >
      Quote
    </p>
    <cite>
      cite
    </cite>
  </blockquote>
</div>
`;
